python-checks/src/main/java/org/sonar/python/checks/UselessParenthesisCheck.java

```diff
@@ -19,93 +19,142 @@
     }
-    return parent.is(PythonGrammar.FOR_STMT) && parent.getFirstChild(PythonGrammar.EXPRLIST).getChildren(PythonGrammar.EXPR).size() == 1;
   }
+   */
 
-  private void checkAtom(AstNode atom, boolean ignoreTestNumber) {
-    if (violationCondition(atom, ignoreTestNumber)) {
-      addIssue(atom, "Remove those useless parentheses");
+  private void checkRedundantParentheses(Token rightParenthesis) {
+    LeftParenthensisData current = leftParenthesisDataStack.peek();
+//    LeftParenthensisData enclosing = leftParenthesisDataStack.get(leftParenthesisDataStack.size() - 2);
+    
+    if (previousTokenWasRightParenthesis && current.getContiguousLeftParenthesis() == last.getLeftParenthesis()) {  
+//      && enclosing.getContiguousLeftParenthesis() == current.getLeftParenthesis()) {
+      addIssue(current.getLeftParenthesis(), MESSAGE).secondary(rightParenthesis, null);
     }
   }
 
-  private static boolean violationCondition(AstNode atom, boolean ignoreTestNumber) {
-    List<AstNode> children = atom.getChildren();
-    boolean result = children.size() == 3 && children.get(0).is(PythonPunctuator.LPARENTHESIS) && children.get(2).is(PythonPunctuator.RPARENTHESIS) && isOnASingleLine(atom);
-    if (result && !ignoreTestNumber) {
-      result = children.get(1).getChildren(PythonGrammar.TEST).size() == 1 && children.get(1).getFirstChild(PythonPunctuator.COMMA) == null;
+  private static class LeftParenthensisData {
+
+    private Token leftParenthesis;
+    
+    private Token rightParenthesis;
+
+    private Token contiguousLeftParenthesis;
+
+    public LeftParenthensisData(Token leftParenthesis) {
+      this.leftParenthesis = leftParenthesis;
     }
-    return result;
-  }
 
-  private void visitNotTest(AstNode node) {
-    boolean hasUselessParenthesis = node.select()
-        .children(PythonGrammar.ATOM)
-        .children(PythonGrammar.TESTLIST_COMP)
-        .children(PythonGrammar.TEST)
-        .children(PythonGrammar.ATOM, PythonGrammar.COMPARISON)
-        .isNotEmpty();
-    if (hasUselessParenthesis) {
-      checkAtom(node.getFirstChild().getNextSibling(), false);
+    public Token getLeftParenthesis() {
+      return leftParenthesis;
     }
-  }
 
-  private static boolean isOnASingleLine(AstNode node) {
-    return node.getTokenLine() == node.getLastToken().getLine();
+    public Token getRightParenthesis() {
+      return rightParenthesis;
+    }
+
+    public Token getContiguousLeftParenthesis() {
+      return contiguousLeftParenthesis;
+    }
+
+    public void setContiguousLeftParenthesis(Token contiguousLeftParenthesis) {
+      this.contiguousLeftParenthesis = contiguousLeftParenthesis;
+    }
+
+    public void setRightParenthesis(Token rightParenthesis) {
+      this.rightParenthesis = rightParenthesis;
+    }
+
+    public boolean isFollowedByLeftParenthesis() {
+      return contiguousLeftParenthesis != null;
+    }
+
+//    public void setFollowedByLeftParenthesis(boolean isFollowedByLeftParenthesis) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)